Fix #21987: Resolve mobile thumbnail modal glitch issue in blog post editor.

ojasvi004 opened the pull request:
## Overview

1. This PR fixes or fixes part of #21987.
2. This PR does the following: This PR fixes a mobile UI issue where the **upload image modal** overlapped the **thumbnail preview** in the blog post editor. Now the modal no longer obstructs the preview.  
3. (For bug-fixing PRs only) The original bug occurred because: The original bug occurred because of a CSS issue where the modal was not properly positioned on small devices. The `@media (max-width: 768px)`  css code resulted in an overlap. This fix adjusts the styling to ensure the modal appears correctly without obstructing the preview.

## Essential Checklist

Please follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Make-a-pull-request).

- [x] I have linked the issue that this PR fixes in the "Development" section of the sidebar.
- [x] I have checked the "Files Changed" tab and confirmed that the changes are what I want to make.
- [ ] I have written tests for my code.
- [x] The **PR title** starts with "Fix #bugnum: " or "Fix part of #bugnum: ...", followed by a short, clear summary of the changes.
- [x] I have assigned the correct reviewers to this PR (or will leave a comment with the phrase "@{{reviewer_username}} PTAL" if I can't assign them directly).

## Proof that changes are correct

[Screencast from 2025-03-01 20-07-47.webm](https://github.com/user-attachments/assets/7044c60b-e6c2-4408-b172-d41e29dd344d)

## PR Pointers

- Never force push! If you do, your PR will be closed.
- To reply to reviewers, follow these instructions: https://github.com/oppia/oppia/wiki/Make-a-pull-request#step-5-address-review-comments-until-all-reviewers-approve
- Some e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR).
- See the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia's-code-owners-and-checks-to-be-carried-out-by-developers) for what code owners will expect.

ojasvi004 committed:
resolve mobile UI glitch in blog post thumbnail upload

ojasvi004 committed:
resolve mobile UI glitch in blog post thumbnail upload

ojasvi004 committed:
resolve mobile UI glitch in blog post thumbnail upload

Nik-09 reviewed (changes requested):
Can we also fix the "Add Thumbnail Image" button size?
![image](https://github.com/user-attachments/assets/e2a7c35f-e836-451d-a908-a9514ce5b694)

Thanks!

ojasvi004 committed:
Merge branch 'oppia:develop' into fix-blogpost-mobile-thumbnail-glitch-21987

ojasvi004 committed:
improve button UI

ojasvi004 commented:
@Nik-09, some checks failed and the error states: 'Expected image to match or be a close match to snapshot but was 11.59% different from snapshot' 
However I have updated the mobile screen UI. How should I proceed? Thanks

previous UI:
![image](https://github.com/user-attachments/assets/b04cb3cf-c553-4e3a-8c71-dbfe082126b4)

updated UI:
![image](https://github.com/user-attachments/assets/4e898ca7-07b9-4359-8d15-d0a0fb2b2690)

ojasvi004 commented:
@Nik-09, PTAL,
I'm facing this issue:

> @Nik-09, some checks failed and the error states: 'Expected image to match or be a close match to snapshot but was 11.59% different from snapshot' However I have updated the mobile screen UI. How should I proceed? Thanks
> 
> previous UI: ![image](https://private-user-images.githubusercontent.com/123376306/420036414-b04cb3cf-c553-4e3a-8c71-dbfe082126b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5NzQ2MzgsIm5iZiI6MTc0MTk3NDMzOCwicGF0aCI6Ii8xMjMzNzYzMDYvNDIwMDM2NDE0LWIwNGNiM2NmLWM1NTMtNGUzYS04YzcxLWRiZmUwODIxMjZiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNFQxNzQ1MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02OGNhYzJiNGMwNzE0MjExNmY3OTgwYjJkYmVjNzE0Y2EyYmQxOTY5NzQ5YTNiM2EyYTI5ZmMwOWNhNTNkM2FkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.FlkFtORbQkY7cEYJmV4jVgFQ5VBYf5Sig2wODlLiJ9U)
> 
> updated UI: ![image](https://private-user-images.githubusercontent.com/123376306/420040722-4e898ca7-07b9-4359-8d15-d0a0fb2b2690.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE5NzQ2MzgsIm5iZiI6MTc0MTk3NDMzOCwicGF0aCI6Ii8xMjMzNzYzMDYvNDIwMDQwNzIyLTRlODk4Y2E3LTA3YjktNDM1OS04ZDE1LWQwYTBmYjJiMjY5MC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxNFQxNzQ1MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yN2VjYzQwNDM4YmNlNDE2ZmI0YTIwMjY2OTQ1NmVkNGVhMDU5ODc3MWI2NDZkMmU4MWI4NGViY2YwODljMzNhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.C1T-ZO17DyO6s9t2_oASUXV7Eb4cYwSbteHOW5oSn-Q)

seanlip commented:
@ojasvi004  -- @imchristie can help you with this. Looping her in here. Thanks.

imchristie commented:
@ojasvi004 To fix the issue you're facing, delete the screenshots `blogEditorPageWithErrorMessageForDuplicateBlogPostTitle` in `blog-editor/prod-mobile-screenshots` and `blog-editor/dev-mobile-screenshots`. Then run the test `blog-editor/try-to-publish-a-duplicate-blog-post-and-get-blocked` in both dev mode and prod mode in mobile environment to regenerate the new screenshots. You should see a new screenshot `blogEditorPageWithErrorMessageForDuplicateBlogPostTitle` in that two folders.

ojasvi004 committed:
update mobile screenshots for blog editor error message on duplicate titles

ojasvi004 committed:
Merge branch 'oppia:develop' into fix-blogpost-mobile-thumbnail-glitch-21987

ojasvi004 commented:
@imchristie, I followed the steps you mentioned, but I'm encountering a similar issue. The test `blog-editor/try-to-publish-a-duplicate-blog-post-and-get-blocked` is failing in the github workflow environment but passes locally on my system. The screenshot shows a 4.77% difference. I suspect this is likely due to different transparency levels in the error message. This probably happened because the screenshots were taken at slightly different moments.

![blogEditorPageWithErrorMessageForDuplicateBlogPostTitle-diff](https://github.com/user-attachments/assets/bcdff27e-990d-4ddd-bc08-f5deaea8f866)

I'm considering downloading the failed images from the github artifacts and using them as new baselines, since the functionality appears correct in both versions. Would this be an acceptable approach or do you recommend another solution?

imchristie commented:
@ojasvi004 For now, only the diff image is available in the diff-snapshot. So how are you able to use them as the baseline? I suggest changing the value of failureTrigger to 0.048 in line 691 in core/tests/puppeteer-acceptance-tests/utilities/common
/puppeteer-utils.ts. It's a temporary solution for now and I will be looking into this problem in the future.

ojasvi004 committed:
adjust failureTrigger increment in puppeteer utils

ojasvi004 commented:
Noted, @imchristie. I had some misunderstanding regarding the diff image layout.
I've made the changes you suggested. Thanks!

ojasvi004 commented:
@imchristie, PTAL. Thanks!

imchristie reviewed (approved):
LGTM!

ojasvi004 commented:
@Nik-09, PTAL. Thanks!

ojasvi004 commented:
@HardikGoyal2003, please merge this PR. Thanks!

ojasvi004 committed:
Merge branch 'develop' into fix-blogpost-mobile-thumbnail-glitch-21987

ojasvi004 committed:
update

ojasvi004 committed:
update

ojasvi004 commented:
@HardikGoyal2003, the PR failed on the develop branch. I've added a commit but there are no actual changes in the code. Could you please merge the PR again? Thanks

HardikGoyal2003 commented:
@ojasvi004  Can you please check this: https://github.com/oppia/oppia/wiki/If-CI-checks-fail-on-your-PR#what-to-do-if-ci-checks-only-fail-in-the-merge-queue

Please file the flake according to the mentioned steps. Thanks!!
